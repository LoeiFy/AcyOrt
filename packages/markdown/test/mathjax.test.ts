import Markdown from '../src'

it('task', () => {
  const m = new Markdown()

  const t0 = `先看一下矩阵的显示效果:
实例：$$f(x)=x$$ 测试
# 安装与配`
  const t1 = `先看一下矩阵的显示效果:
实例：$f(x)=x$
# 安装与配`
  const t2 = `先看一下矩阵的显示效果:
实例：$$f(x)=x$
# 安装与配`
  const t3 = `先看一下矩阵的显示效果:
实例：$f(x)=x$$
# 安装与配`
  const t4 = 'aaa'

  expect(m.render(t0)).toMatchSnapshot()
  expect(m.render(t1)).toMatchSnapshot()
  expect(m.render(t2)).toMatchSnapshot()
  expect(m.render(t3)).toMatchSnapshot()
  expect(m.render(t4)).toMatchSnapshot()
})
