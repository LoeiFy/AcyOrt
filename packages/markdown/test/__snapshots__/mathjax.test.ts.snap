// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`task 1`] = `
"<p>先看一下矩阵的显示效果:
实例：</p>
  <div class="latex-math" style="text-align:center">
    <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mo stretchy="false">(</mo><mi>x</mi><mo stretchy="false">)</mo><mo>=</mo><mi>x</mi></mrow><annotation encoding="application/x-tex">f(x)=x</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">x</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:0.4306em;"></span><span class="mord mathnormal">x</span></span></span></span>
  </div>
<p> 测试</p><h1>
  <a href="#安装与配" id="安装与配"></a>安装与配
</h1>"
`;

exports[`task 2`] = `
"<p class="latex-math">先看一下矩阵的显示效果:
实例：<span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mo stretchy="false">(</mo><mi>x</mi><mo stretchy="false">)</mo><mo>=</mo><mi>x</mi></mrow><annotation encoding="application/x-tex">f(x)=x</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">x</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:0.4306em;"></span><span class="mord mathnormal">x</span></span></span></span></p><h1>
  <a href="#安装与配" id="安装与配"></a>安装与配
</h1>"
`;

exports[`task 3`] = `
"<p>先看一下矩阵的显示效果:
实例：$$f(x)=x$</p><h1>
  <a href="#安装与配" id="安装与配"></a>安装与配
</h1>"
`;

exports[`task 4`] = `
"<p class="latex-math">先看一下矩阵的显示效果:
实例：<span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mo stretchy="false">(</mo><mi>x</mi><mo stretchy="false">)</mo><mo>=</mo><mi>x</mi></mrow><annotation encoding="application/x-tex">f(x)=x</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">x</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:0.4306em;"></span><span class="mord mathnormal">x</span></span></span></span>$</p><h1>
  <a href="#安装与配" id="安装与配"></a>安装与配
</h1>"
`;

exports[`task 5`] = `"<p>aaa</p>"`;
